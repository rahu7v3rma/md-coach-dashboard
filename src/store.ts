import { configureStore } from '@reduxjs/toolkit';

import clientReducer from './reducers/client';
import groupReducer from './reducers/group';
import notificationReducer from './reducers/notification';
import userReducer from './reducers/user';

// define store options as a variable so we can export it for tests to use
const storeOptions: any = {
    reducer: {
        user: userReducer,
        client: clientReducer,
        notification: notificationReducer,
        group: groupReducer
    }
};

const store = configureStore(storeOptions);

export default store;

// infer the `RootState` type from the store
export type RootState = ReturnType<typeof store.getState>;

// infer the `AppDispatch` type from the store so thunk actions can return promises
export type AppDispatch = typeof store.dispatch;

export const test = {
    storeOptions
};
