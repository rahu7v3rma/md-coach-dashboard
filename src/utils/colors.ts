const PRIMARY = '#72996B';
const PRIMARY_LIGHT = '#D3E29F';
const PRIMARY_LIGHTER = 'rgba(211, 226, 159, 0.25)';
const PRIMARY_LIGHTEST = '#FAFAFF';
const SECONDARY = '#F0E957';
const BLACK = '#000000';
const BLACK_02 = 'rgba(0, 0, 0, 0.2)';
const BLACK_04 = 'rgba(0, 0, 0, 0.4)';
const BLACK_1 = 'rgba(0, 0, 0, 0.05)';
const BLACK_2 = 'rgba(0, 0, 0, 0.06)';
const WHITE = '#FFFFFF';
const LIGHT_2 = '#ECF8E6';
const LIGHTER = '#D3E29F40';
const GRAY_LIGHTEST = '#F8F8F8';
const GRAY = '#A4AAAF';
const BLACK_TEXT = '#271A51';
const LIGHT_SKY_BLUE = '#D3E6F8';
const BLACK_LIGHT_TEXT = 'rgba(77, 79, 77, 1)';
const GREEN_TEXT = 'rgba(114, 153, 107, 1)';
const BORDER_COLOR = '#D3E6F8';
const BORDER_BOTTOM_COLOR = 'rgba(248, 248, 248, 1)';
const SUB_TITLE_TXET = '#A4AAAF';
const DEVIDER_COLOR = '#d3e6f8';
const WHITE_1 = '#F5F8FB';
const ERROR = '#B16663';
const GREEN = '#3a9a00';
const SILVER = '#D6D6D6';
const SLATE_BLUE = '#6e51d0';
const OCEAN_BLUE = '#5a3dbf';
const ALICE_BLUE = '#eef4fa';
const KELLY_GREEN = '#4DBA0B';
const CITRINE = '#DCD40F';
const DARK_TERRA_COTTA = '#D05151';
const SILVER_SAND = '#babfc3';
const BLUE_JEANS = '#59a2ff';
const DAVYS_GREY = '#58595a';
const GAINSBORO = '#DADADA';
const RED = '#ff0000';
const CHINESE_WHITE = '#E7D8EE';
const LINEN = '#FFECEC';
const CYAN_LIGHT = '#D4F9F9';
const IRIS = '#5652BF';
const DARK_LIVER = '#4D4F4D';
const ORANGE = '#FF9330';
const YELLOW = '#DCD40F';
const LIGHT_GREEN = '#D6E1D8';
const LATEST_NOTE_BACKGROUND = '#F9FBF9';
const NOTE_BACKGROUND = '#F9F9FE';
const MAIN_BACKGROUND = '#F5F4EE';
const EARLY_DAWN = '#FFF9DB';
const LIGHT_SKY_BLUE_2 = '#3ED1FF';
const LIGHT_PINK = '#E19EDF';
const LIGHT_GREEN_2 = '#B3E197';

export const Colors = {
    theme: {
        primary: PRIMARY,
        secondary: SECONDARY,
        gray: GRAY,
        lightSkyBlue: LIGHT_SKY_BLUE,
        primaryLight: PRIMARY_LIGHT,
        primaryLighter: PRIMARY_LIGHTER,
        primaryLightest: PRIMARY_LIGHTEST
    },
    extra: {
        white: WHITE,
        black: BLACK,
        light2: LIGHT_2,
        lighter: LIGHTER,
        greyLightest: GRAY_LIGHTEST,
        blackText: BLACK_TEXT,
        offWhite: PRIMARY_LIGHTEST,
        black_text: BLACK_TEXT,
        black_light_text: BLACK_LIGHT_TEXT,
        green_text: GREEN_TEXT,
        border_color: BORDER_COLOR,
        border_bottom_color: BORDER_BOTTOM_COLOR,
        sub_title_text: SUB_TITLE_TXET,
        divider_color: DEVIDER_COLOR,
        off_white: PRIMARY_LIGHTEST,
        white1: WHITE_1,
        black1: BLACK_1,
        black2: BLACK_2,
        black02: BLACK_02,
        black04: BLACK_04,
        error: ERROR,
        orange: ORANGE,
        green: GREEN,
        silver: SILVER,
        slateBlue: SLATE_BLUE,
        oceanBlue: OCEAN_BLUE,
        aliceBlue: ALICE_BLUE,
        kellyGreen: KELLY_GREEN,
        citrine: CITRINE,
        darkTerraCotta: DARK_TERRA_COTTA,
        silverSand: SILVER_SAND,
        blueJeans: BLUE_JEANS,
        davysGrey: DAVYS_GREY,
        gainsboro: GAINSBORO,
        red: RED,
        chineseWhite: CHINESE_WHITE,
        linen: LINEN,
        cyanLight: CYAN_LIGHT,
        iris: IRIS,
        darkLiver: DARK_LIVER,
        yellow: YELLOW,
        green_light: LIGHT_GREEN,
        latest_note_background: LATEST_NOTE_BACKGROUND,
        note_background: NOTE_BACKGROUND,
        main_background: MAIN_BACKGROUND,
        earlyDawn: EARLY_DAWN,
        skyBlue2: LIGHT_SKY_BLUE_2,
        lightPink: LIGHT_PINK,
        lightGreen2: LIGHT_GREEN_2
    }
};
