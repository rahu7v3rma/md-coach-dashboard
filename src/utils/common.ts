export const BASE_API_URL = process.env.REACT_APP_BACKEND_URL || '/';

export const COMMON = {
    stringFormat(s: string, ...args: any[]) {
        return s.replace(/{([0-9]+)}/g, (match, index) =>
            typeof args[index] === 'undefined' ? match : args[index]
        );
    }
};
