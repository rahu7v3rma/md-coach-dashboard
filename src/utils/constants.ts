import Activity from 'src/assets/activity.svg';
import Drop from 'src/assets/drop.svg';
import Food from 'src/assets/food.svg';
import Hydration from 'src/assets/hydrationDrop.svg';
import Insulin from 'src/assets/insulin.svg';
import Medication from 'src/assets/medication.svg';
import Timer from 'src/assets/timer.svg';
import Weight from 'src/assets/weight.svg';
import { Log } from 'src/types/log';

export const logTypeDetails: Record<string, Log> = {
    UserExercise: {
        icon: Activity,
        title: 'Activity'
    },
    UserWeight: {
        icon: Weight,
        title: 'Weight'
    },
    UserFast: {
        icon: Timer,
        title: 'Begin fast'
    },
    UserInsulin: {
        icon: Insulin,
        title: 'Insulin'
    },
    UserGlucose: {
        icon: Drop,
        title: 'Glucose'
    },
    UserDrink: {
        icon: Hydration,
        title: 'Hydration'
    },
    UserMedication: {
        icon: Medication,
        title: 'Medication'
    },
    UserFood: {
        icon: Food,
        title: 'Food'
    },
    UserLesson: {
        icon: Activity,
        title: 'Lesson Completed'
    }
};

export const NOTIFICATION_TYPE = {
    STREAM_CHAT_MESSAGE: 'message.new',
    LESSON_COMPLETED: 'lesson.completed'
};

export const CLIENTS_TABLE_COLUMNS = [
    'name',
    'type',
    'engagement',
    'fbg',
    'last_weight',
    'last_week_avg_fbg',
    'activity_type',
    'duration',
    'weekly_total',
    'hydration'
];

export const timeFrames = [
    { label: 'Last 7 days', value: '7' },
    { label: 'Last 14 days', value: '14' },
    { label: 'Last 30 days', value: '30' }
];
