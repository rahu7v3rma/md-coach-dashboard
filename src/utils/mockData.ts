import moment from 'moment';

import { Colors } from './colors';

export const chatUserList = [
    {
        id: 'e455062a4ec9ccef9a3d651fb9b2c7b25f983dbcc9c90be976b0996bcbd3cc6c',
        role: 'customer',
        created_at: '2023-09-05T12:44:09.740891Z',
        updated_at: '2023-09-07T15:58:44.949379Z',
        last_active: '2023-09-07T19:04:36.836449Z',
        banned: false,
        online: true,
        shadow_banned: false,
        name: 'A B',
        image: null
    },
    {
        id: '52c94d3034d2c4defcf2997eb562724a9af015df0ae42da110b90f7267ac8659',
        role: 'customer',
        created_at: '2023-09-04T07:58:04.22538Z',
        updated_at: '2023-09-05T12:40:11.565452Z',
        last_active: moment().subtract(1, 'week'),
        banned: false,
        online: false,
        shadow_banned: false,
        image: null
    },
    {
        id: 'ed4d83c5e5c15be2a8afd2f18743dd013a59d9fb29470d0a6cb2324e72d03960',
        role: 'guest',
        created_at: '2023-09-01T09:18:19.111097Z',
        updated_at: '2023-09-04T07:51:43.157236Z',
        last_active: '2023-09-04T07:45:47.402292Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'a b',
        image: null
    },
    {
        id: '27b588063af6f83b735e3619c60149d33e45b4966bf6deb9e2670598fcc5bbca',
        role: 'customer',
        created_at: '2023-09-01T09:17:44.66015Z',
        updated_at: '2023-09-01T09:17:44.838758Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'a b',
        image: null
    },
    {
        id: '27c6fe5289a404a80ba9d5a27d06cce6cae3ea4157a010c07d868aebaa23afd1',
        role: 'customer',
        created_at: '2023-08-31T10:31:07.072598Z',
        updated_at: '2023-09-01T09:16:42.771103Z',
        last_active: '2023-09-01T09:16:42.822099Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Omry Ka',
        image: null
    },
    {
        id: 'fde7e986515a608e8204c5bfcc45361a96f0298bd20088fc2cd462cad799c5e3',
        role: 'coach',
        created_at: '2023-08-28T10:27:12.228277Z',
        updated_at: '2023-08-29T11:29:55.961657Z',
        last_active: '2023-08-29T12:32:19.47458Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Coach Yaron Coach',
        image: null
    },
    {
        id: '8a7babbaa73795bc612660ed2fc2181d7840fc36f01ad2adb654feeba876bdb4',
        role: 'customer',
        created_at: '2023-08-28T09:33:02.880221Z',
        updated_at: '2023-09-01T09:02:01.189652Z',
        last_active: '2023-09-01T09:02:01.066318Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'a a',
        image: null
    },
    {
        id: '9ae174a646533bca03fd789ec458bd95091897506f1e1f075a13fc2800b166bf',
        role: 'customer',
        created_at: '2023-08-28T09:33:01.299363Z',
        updated_at: '2023-08-28T09:33:01.313714Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'a a',
        image: null
    },
    {
        id: 'dd5f523d62dd0d44d2003723ee4c6d9d3005eb32793f147a8a69155becd421c7',
        role: 'customer',
        created_at: '2023-08-21T13:17:09.950661Z',
        updated_at: '2023-08-31T09:08:41.160561Z',
        last_active: '2023-08-31T09:06:47.442304Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'A B',
        image: null
    },
    {
        id: 'a575da8fdab1adeebc2dcfc926ce63d73066bf1df3bfbee9404fc28da425ceac',
        role: 'customer',
        created_at: '2023-08-21T12:47:56.941808Z',
        updated_at: '2023-08-21T13:15:15.226448Z',
        last_active: '2023-08-21T13:13:39.187452Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'a b',
        image: null
    },
    {
        id: '108287176418a6878323b0b450ef05f633de56591795eb5c259435d5dfc0c2ee',
        role: 'customer',
        created_at: '2023-08-20T11:36:37.725694Z',
        updated_at: '2023-09-15T15:18:08.065261Z',
        last_active: '2023-09-15T15:17:55.832476Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'a b',
        image: null
    },
    {
        id: '7f1ce02a7222cb34e266ea1f93024144f2792d00c6e7948e61171bad7450a439',
        role: 'customer',
        created_at: '2023-08-20T11:25:59.750747Z',
        updated_at: '2023-08-20T11:27:30.988385Z',
        last_active: '2023-08-20T11:26:53.694128Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'OK KA',
        image: null
    },
    {
        id: '89e8cba16afd3ca204f6421546e5ec1bd64edebc60d3a209b22ee0a4fa2312f3',
        role: 'customer',
        created_at: '2023-08-20T10:52:47.637028Z',
        updated_at: '2023-08-20T11:23:45.532031Z',
        last_active: '2023-08-20T11:14:39.3116Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Omry K',
        image: null
    },
    {
        id: '212c41a447b737d8c7d9c4fd399c4c834be7e0d7980fbc7f46862e7b5927f055',
        role: 'customer',
        created_at: '2023-08-07T05:55:52.220389Z',
        updated_at: '2023-08-17T04:55:34.617996Z',
        last_active: '2023-08-17T04:55:34.764832Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Om Ka',
        image: null
    },
    {
        id: '9beeeed2cd8276b9d3a857e7bbc88f670fef5b7b52ed51c1f32a6da685392e39',
        role: 'customer',
        created_at: '2023-07-27T12:43:14.660127Z',
        updated_at: '2023-07-27T12:43:14.847233Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Mor Customer #42',
        image: null
    },
    {
        id: 'fc241a97eaeaacce52137b71b1cf0c9687b7ad9c9ed0e2663f1a131f299b7eb0',
        role: 'customer',
        created_at: '2023-07-27T12:38:49.303555Z',
        updated_at: '2023-09-14T20:39:44.397305Z',
        last_active: '2023-09-14T20:36:30.298281Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Mor Customer #41',
        image: null
    },
    {
        id: 'eef10747d83ca13adc64195e2f31f7ba20b024675425078cadfd396d22e288b8',
        role: 'customer',
        created_at: '2023-07-26T17:00:49.063803Z',
        updated_at: '2023-07-26T17:00:49.257857Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Mor Customer #40',
        image: null
    },
    {
        id: 'f44a8a0b0d3a7cdd63849ad1ae12115a95e97627bbafce0a98584c3adc880064',
        role: 'customer',
        created_at: '2023-07-26T16:56:27.365582Z',
        updated_at: '2023-07-27T12:38:16.246013Z',
        last_active: '2023-07-27T12:26:35.725764Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Mor Customer #39',
        image: null
    },
    {
        id: '9fbdda42254526fba4f78adc15eea396399a95a13fae8f54f3eb3fdd6d25f6d9',
        role: 'customer',
        created_at: '2023-07-26T16:51:18.398265Z',
        updated_at: '2023-07-26T16:51:18.576979Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Mor Customer #38',
        image: null
    },
    {
        id: '806f40a10af600c503e25ffc17837e5d832239a2f5de12346cd0a44292ab7aa2',
        role: 'customer',
        created_at: '2023-07-26T16:23:40.226631Z',
        updated_at: '2023-07-26T16:23:40.409256Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Mor Custoemr #37',
        image: null
    },
    {
        id: 'b60a96e7e4b71e2d806ff3706b7ca76cbe598aaf33f06acef5feb59802dd04ab',
        role: 'customer',
        created_at: '2023-07-26T16:22:54.577357Z',
        updated_at: '2023-07-26T16:22:54.762307Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Mor Customer #36',
        image: null
    },
    {
        id: 'b7fd9a8f8499280da04e5573067ee291419bd99fa7dac56925e53d4eb0d4071e',
        role: 'customer',
        created_at: '2023-07-26T16:20:58.300678Z',
        updated_at: '2023-07-26T16:20:58.486555Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Mor Customer #35',
        image: null
    },
    {
        id: 'ac6807255d5565787288a5b2696477a6d5d2f11bef780297608b6f93f9e784df',
        role: 'customer',
        created_at: '2023-07-24T08:04:45.787141Z',
        updated_at: '2023-07-24T08:13:50.30187Z',
        last_active: '2023-07-24T08:06:27.21234Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Omry Imperial',
        image: null
    },
    {
        id: 'b24177b059a56b9a1d4b1f2a57df978a715e3fc16d9b7463c0f6d3fb1519d0a5',
        role: 'customer',
        created_at: '2023-07-24T08:03:32.113199Z',
        updated_at: '2023-07-26T08:52:37.166469Z',
        last_active: '2023-07-26T11:41:09.381716Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Omry Metirc',
        image: null
    },
    {
        id: 'b5010f10e72a466e5cc2e54cc18127b983f01b2c24aa04ad838365248c5a8c46',
        role: 'customer',
        created_at: '2023-07-24T07:45:04.527658Z',
        updated_at: '2023-07-24T07:45:04.942188Z',
        last_active: '2023-07-24T07:45:05.326495Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Omry3 Kaufman3',
        image: null
    },
    {
        id: '547dfbcc1952695fda6b23322390657a1e42989b01c7192e0d36914a2bfe1499',
        role: 'customer',
        created_at: '2023-07-24T07:42:32.12563Z',
        updated_at: '2023-07-24T07:42:32.609928Z',
        last_active: '2023-07-24T07:42:33.014716Z',
        banned: false,
        online: false,
        shadow_banned: false,
        image: null,
        name: 'Omry2 Kaufman2'
    },
    {
        id: 'f55d9d4359f766d235d0b519c19215e95161b989c1a8ff19f663cdd61fb4c92f',
        role: 'customer',
        created_at: '2023-07-24T07:41:02.274611Z',
        updated_at: '2023-07-24T07:41:02.687083Z',
        last_active: '2023-07-24T07:41:03.076568Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'omry kaufman',
        image: null
    },
    {
        id: 'be7aa7a8a6e8e3f0b3969f9082f8ad5fa76bf030e1f6c96e71095d0bc33e016c',
        role: 'customer',
        created_at: '2023-07-24T07:40:03.665105Z',
        updated_at: '2023-07-24T07:40:03.988971Z',
        last_active: '2023-07-24T07:40:05.172638Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'omry kaufman',
        image: null
    },
    {
        id: 'e5ed8d7ead2d567377f4222994823b270c28ea9675410377c68a88521a061145',
        role: 'customer',
        created_at: '2023-07-22T17:56:16.197789Z',
        updated_at: '2023-07-22T17:56:16.368887Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Mor Customer #34',
        image: null
    },
    {
        id: 'b12082b684bd775846d36873596a0e965d76a2238d00b0a41a7fe891f21392c9',
        role: 'customer',
        created_at: '2023-07-21T20:12:27.492236Z',
        updated_at: '2023-07-23T20:54:14.413015Z',
        last_active: '2023-07-23T20:43:05.192208Z',
        banned: false,
        online: false,
        shadow_banned: false,
        name: 'Mor Customer #33',
        image: null
    }
];

export const userClient = [
    {
        id: 1,
        name: 'Justin Curtis',
        type: 'T2D',
        group: 'Group B',
        engagement: '2023-04-06T12:23:00Z',
        last_fasting: 117,
        second_last_fasting: 117,
        last_weight: 142,
        second_last_weight: 142,
        weight_units: 'kg',
        last_week_avg_fbg: 142,
        activity_type: 'Running',
        duration: 35,
        weekly_total: 125,
        hydration: 3,
        hydration_unit: 'cups',
        fbg: 120,
        second_last_fbg: 130,
        fbg_unit: 'mmol',
        weight_unit: 'kg'
    },
    {
        id: 2,
        name: 'Allison Franci',
        type: 'PreDB',
        group: 'Group A',
        engagement: '2023-04-01T12:23:00Z',
        last_fasting: 88,
        second_last_fasting: 117,
        last_weight: 194,
        second_last_weight: 142,
        weight_units: 'kg',
        last_week_avg_fbg: 142,
        activity_type: 'Walking',
        duration: 35,
        weekly_total: 125,
        hydration: 2,
        hydration_unit: 'cups',
        fbg: 12,
        second_last_fbg: 8,
        fbg_unit: 'mmol',
        weight_unit: 'kg'
    }
];

export const user = {
    id: 'e455062a4ec9ccef9a3d651fb9b2c7b25f983dbcc9c90be976b0996bcbd3cc6c',
    role: 'customer',
    created_at: '2023-09-05T12:44:09.740891Z',
    updated_at: '2023-09-07T15:58:44.949379Z',
    last_active: '2023-09-07T19:04:36.836449Z',
    banned: false,
    online: false,
    shadow_banned: false,
    name: 'A B',
    image: 'image.com'
};
export const currentMembers = [
    'a5de9604d556d6f190a23e5a7dffb7a49c3ada560258fdc3e48c0c98c0533ea8',
    'ab2e30bf75636a9c97e05099d9ccc9b623c17a44f629e7ce7206dba683028db1',
    'e0bc614e4fd035a488619799853b075143deea596c477b8dc077e309c0fe42e9'
];

const clientDetailsAnalyticsCardsChartData = [
    { x: '01/02/2024', y: 100 },
    { x: '02/02/2024', y: 200 },
    { x: '03/02/2024', y: 300 },
    { x: '04/02/2024', y: 400 },
    { x: '05/02/2024', y: 500 }
];
export const clientDetailsAnalyticsCards = [
    {
        title: 'Glucose',
        color: Colors.extra.red,
        chart: {
            type: 'line' as 'line',
            data: [
                { x: '01/02/2024', y: 100 },
                { x: '02/02/2024', y: 200 },
                { x: '03/02/2024', y: null },
                { x: '04/02/2024', y: 300 },
                { x: '05/02/2024', y: 400 }
            ]
        }
    },
    {
        title: 'Weight (kg)',
        color: Colors.extra.slateBlue,
        chart: {
            type: 'line' as 'line',
            data: clientDetailsAnalyticsCardsChartData
        }
    },
    {
        title: 'Fasting',
        color: Colors.extra.orange,
        chart: {
            type: 'bar' as 'bar',
            data: clientDetailsAnalyticsCardsChartData
        }
    },
    {
        title: 'Insulin',
        color: Colors.extra.green,
        chart: {
            type: 'bar' as 'bar',
            data: clientDetailsAnalyticsCardsChartData
        }
    }
];
