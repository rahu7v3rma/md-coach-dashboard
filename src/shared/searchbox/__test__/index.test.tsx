import 'jest-styled-components';
import React from 'react';
import { Provider } from 'react-redux';
import * as router from 'react-router';
import { BrowserRouter } from 'react-router-dom';
import { act, create } from 'react-test-renderer';

import SearchBox, { SearchBoxContainer, SearchIcon } from '..';
import Icon from '../../../assets/search-icon-green.svg';
import store from '../../../store';
import { Colors } from 'src/utils/colors';

const navigate = jest.fn();

describe('SearchBox', () => {
    beforeEach(() => {
        jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
    });

    it('SearchBox snapshot', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <SearchBox />
                </BrowserRouter>
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });

    it('SearchBox renders without errors', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <SearchBox />
                </BrowserRouter>
            </Provider>
        );
        expect(tree.root).toBeTruthy();
    });

    it('renders the search icon and input element', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <SearchBox />
                </BrowserRouter>
            </Provider>
        );
        let icon = tree.root.findByProps({
            src: Icon
        }).props;
        expect(icon).toBeTruthy();
        let input = tree.root.findByProps({ placeholder: 'Search' }).props;
        expect(input).toBeTruthy();
    });

    it('calls the onChange callback when text is typed', async () => {
        const onChange = jest.fn();
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <SearchBox onChange={onChange} />
                </BrowserRouter>
            </Provider>
        );
        let input = tree.root.findByProps({ placeholder: 'Search' }).props;
        await act(() => input.onChange({ target: { value: '1' } }));
        expect(onChange).toHaveBeenCalledTimes(1);
    });

    it('applies styles correctly to the search box', async () => {
        const tree = create(<SearchBoxContainer />).toJSON();
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('background', Colors.extra.white);
        expect(tree).toHaveStyleRule('width', '350px');
        expect(tree).toHaveStyleRule('padding', '19px 23px');
        expect(tree).toHaveStyleRule('border-radius', '16px');
    });

    it('applies styles correctly to the search icon', async () => {
        const tree = create(<SearchIcon />).toJSON();
        expect(tree).toHaveStyleRule('width', '18px');
        expect(tree).toHaveStyleRule('height', '18px');
        expect(tree).toHaveStyleRule('margin-right', '11px');
    });

    it('calls onChange with the correct value when text is typed', async () => {
        const onChange = jest.fn();
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <SearchBox onChange={onChange} />
                </BrowserRouter>
            </Provider>
        );
        let input = tree.root.findByProps({ placeholder: 'Search' }).props;
        await act(() => input.onChange({ target: { value: 'jest' } }));
        expect(onChange).toHaveBeenCalledWith('jest');
    });

    it('does not call onChange when text is not typed', async () => {
        const onChange = jest.fn();
        create(
            <Provider store={store}>
                <BrowserRouter>
                    <SearchBox onChange={onChange} />
                </BrowserRouter>
            </Provider>
        );
        expect(onChange).toHaveBeenCalledTimes(0);
    });
});
