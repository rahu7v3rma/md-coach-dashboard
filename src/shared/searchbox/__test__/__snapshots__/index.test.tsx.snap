// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox SearchBox snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #FFFFFF;
  width: 350px;
  padding: 19px 23px;
  border-radius: 16px;
  box-shadow: 0px 2px 17px rgba(0,0,0,0.05);
}

.c1 {
  width: 18px;
  height: 18px;
  margin-right: 11px;
}

.c2 {
  font: 500 16px/22px 'Poppins',sans-serif;
  border: 0;
  width: 100%;
}

.c2::-webkit-input-placeholder {
  font: 500 16px/22px 'Poppins',sans-serif;
}

.c2::-moz-placeholder {
  font: 500 16px/22px 'Poppins',sans-serif;
}

.c2:-ms-input-placeholder {
  font: 500 16px/22px 'Poppins',sans-serif;
}

.c2::placeholder {
  font: 500 16px/22px 'Poppins',sans-serif;
}

.c2:focus {
  outline: none;
}

<div
  className="c0"
>
  <img
    className="c1"
    src={Object {}}
  />
  <input
    className="c2"
    onChange={[Function]}
    placeholder="Search"
  />
</div>
`;
